sdk/tests/conformance/extensions/webgl-compressed-texture-astc.html

```diff
@@ -1588,6 +1588,35 @@
 var canvas = document.getElementById("canvas");
 var gl = wtu.create3DContext(canvas, {antialias: false});
 var program = wtu.setupTexturedQuad(gl);
+
+// For LDR and non-sRGB formats, the error color is magenta *or* all NaNs. To
+// make sure we accept this we create a program that transforms an all NaN
+// sampling result to magenta.
+var nanToMagentaProgram = wtu.setupTexturedQuad(gl);
+var fragmentShader = null;
+for (let shader of gl.getAttachedShaders(nanToMagentaProgram)) {
+    if (gl.getShaderParameter(shader, gl.SHADER_TYPE) == gl.FRAGMENT_SHADER) {
+        fragmentShader = shader;
```

Comment:
I don't really like patching the program with a different shader like this, but ok.

Reply:
I also think it'd be nicer to explicitly set up the program the intended way the first time.